### PR TITLE
clarify boxing conversions

### LINKED ISSUE #17335: Value types are objects!
Hey Bill and the rest of the team, 

I am not super happy with how this page treats "object". It strongly implies that variables of value type are not "real" objects, but they are; you can call methods on them without boxing, for instance. It's not the case that in C# an int "becomes an object on demand". It already is an object; values of value type are objects in the C# type system. (The C# type system is of course implemented on top of the CLR's type system, which does make a stronger distinction in the type system, but this documentation is about the C# type system, not its implementation details.)

A more accurate way to describe the C# type system here is that *sometimes you need to convert a value of value type to a value of a reference type, and in those cases the value will automatically be boxed*.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b9770af4-de82-7034-a56b-c68a6b5c55e5
* Version Independent ID: 6bd64fcc-5c84-e541-12f4-0cdcfe4f5dd2
* Content: [C# Types and Variables - A tour of the C# language](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/types-and-variables)
* Content Source: [docs/csharp/tour-of-csharp/types-and-variables.md](https://github.com/dotnet/docs/blob/master/docs/csharp/tour-of-csharp/types-and-variables.md)
* Product: **dotnet-csharp**
* Technology: **csharp-get-started**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #17335

The original explanation implied that value types aren't objects. Modify the language in the two paragraphs on boxing and unboxing. The new language should suggest that value types are being assigned to object references, not converted.

### CODE REVIEW

File: docs/csharp/tour-of-csharp/types-and-variables.md [hunk 1]
```diff
@@ -78,9 +78,9 @@ C#’s type system is unified such that a value of any type can be treated as an
 
 [!code-csharp[Boxing](../../../samples/snippets/csharp/tour/types-and-variables/Program.cs#L1-L10)]
 
-When a value of a value type is converted to type `object`, an `object` instance, also called a "box", is allocated to hold the value, and the value is copied into that box. Conversely, when an `object` reference is cast to a value type, a check is made that the referenced `object` is a box of the correct value type, and, if the check succeeds, the value in the box is copied out.
+When a value of a value type is assigned to an `object` reference, an `object` instance, also called a "box", is allocated to hold the value, and the value is copied into that box. Conversely, when an `object` reference is cast to a value type, a check is made that the referenced `object` is a box of the correct value type. If the check succeeds, the value in the box is copied to the value type.
```
Comment (tdykstra) [inline docs/csharp/tour-of-csharp/types-and-variables.md:81]: Here's a stab at avoiding the construct "an `object` reference, an `object` instance," which is a little hard to follow at first.
```suggestion
When a value of a value type is assigned to an `object` reference, an `object` instance is allocated to hold the value . The `object` instance is called a box, and the value is copied into that box. Conversely, when an `object` reference is cast to a value type, a check is made that the referenced `object` is a box of the correct value type. If the check succeeds, the value in the box is copied to the value type.
```

**Review by tdykstra — APPROVED** [review]
Good clarification, I had one suggestion for a slight change in sentence construction.

File: docs/csharp/tour-of-csharp/types-and-variables.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/tour-of-csharp/types-and-variables.md]: I like that, and then modified the wording a bit more.
